pkg/util/push/otel.go

```diff
@@ -150,10 +152,40 @@ func OTLPHandler(
 		)
 
 		req.Timeseries = metrics
+
+		metadata := otelMetricsToMetadata(otlpReq.Metrics())
+		req.Metadata = metadata
+
 		return body, nil
 	})
 }
 
+func otelMetricsToMetadata(md pmetric.Metrics) []*mimirpb.MetricMetadata {
+	var metadata []*mimirpb.MetricMetadata
+
+	resourceMetricsSlice := md.ResourceMetrics()
+	for i := 0; i < resourceMetricsSlice.Len(); i++ {
+		resourceMetrics := resourceMetricsSlice.At(i)
+		scopeMetricsSlice := resourceMetrics.ScopeMetrics()
+
+		for j := 0; j < scopeMetricsSlice.Len(); j++ {
+			scopeMetrics := scopeMetricsSlice.At(j)
+			for k := 0; k < scopeMetrics.Metrics().Len(); k++ {
+				entry := mimirpb.MetricMetadata{
+					Type:             mimirpb.MetricMetadata_MetricType(scopeMetrics.Metrics().At(k).Type()),
+					MetricFamilyName: prometheustranslator.BuildCompliantName(scopeMetrics.Metrics().At(k), "", true), // TODO expose addMetricSuffixes in configuration
+					Help:             scopeMetrics.Metrics().At(k).Description(),
+					Unit:             scopeMetrics.Metrics().At(k).Unit(),
```

Comment:
I'd suggest extracting `scopeMetrics.Metrics().At(k)` out to a variable to reduce the duplication here.

Reply:
Done.